src/test/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/integration/MigrationControllerIntTest.kt

```diff
@@ -0,0 +1,333 @@
+      val authorizationConsent =
+        verifyAuthorities(client.id!!, client.clientId, "ROLE_CURIOUS_API", "ROLE_VIEW_PRISONER_DATA", "ROLE_COMMUNITY")
+
+      val clientDeployment = clientDeploymentRepository.findById("testz").get()
+      assertThat(clientDeployment.baseClientId).isEqualTo("testz")
+      assertThat(clientDeployment.clientType).isEqualTo(ClientType.PERSONAL)
+      assertThat(clientDeployment.team).isEqualTo("testing team")
+      assertThat(clientDeployment.teamContact).isEqualTo("testz lead")
+      assertThat(clientDeployment.teamSlack).isEqualTo("#testz")
+      assertThat(clientDeployment.hosting).isEqualTo(Hosting.CLOUDPLATFORM)
+      assertThat(clientDeployment.namespace).isEqualTo("testz-testing-dev")
+      assertThat(clientDeployment.deployment).isEqualTo("hmpps-testing-dev")
+      assertThat(clientDeployment.secretName).isEqualTo("hmpps-testing")
+      assertThat(clientDeployment.clientIdKey).isEqualTo("SYSTEM_CLIENT_ID")
+      assertThat(clientDeployment.secretKey).isEqualTo("SYSTEM_CLIENT_SECRET")
+
+      verify(telemetryClient).trackEvent(
+        "AuthorizationServerDetailsMigrate",
+        mapOf("username" to "AUTH_ADM", "clientId" to "testz"),
+        null,
+      )
+
+      clientRepository.delete(client)
+      clientConfigRepository.delete(clientConfig)
+      authorizationConsentRepository.delete(authorizationConsent)
+    }
+
+    @Test
+    fun `register incomplete client`() {
+      assertNull(clientRepository.findClientByClientId("testp"))
+
+      webTestClient.post().uri("/migrate-client")
+        .headers(setAuthorisation(roles = listOf("ROLE_OAUTH_ADMIN")))
+        .body(
+          BodyInserters.fromValue(
+            mapOf(
+              "clientId" to "testp",
+              "clientSecret" to "clientSecret",
+              "clientIdIssuedAt" to "2021-11-25T14:20:00Z",
+            ),
+          ),
+        )
+        .exchange()
+        .expectStatus().isOk
+
+      val client = clientRepository.findClientByClientId("testp")
+
+      assertNotNull(client)
+      assertThat(client!!.clientId).isEqualTo("testp")
+      assertThat(client.clientName).isEqualTo("testp")
+      assertThat(client.clientSecret).isEqualTo("{bcrypt}clientSecret")
+      assertThat(client.authorizationGrantTypes).isEqualTo(AuthorizationGrantType.CLIENT_CREDENTIALS.value)
+      assertThat(client.scopes).containsOnly("read")
+      assertFalse(clientConfigRepository.findById(client.clientId).isPresent)
+      assertFalse(authorizationConsentRepository.findById(AuthorizationConsent.AuthorizationConsentId(client.id, client.clientId)).isPresent)
+
+      verify(telemetryClient).trackEvent(
+        "AuthorizationServerDetailsMigrate",
+        mapOf("username" to "AUTH_ADM", "clientId" to "testp"),
+        null,
+      )
+
+      clientRepository.delete(client)
+    }
+
+    @Test
+    fun `omit mandatory fields gives bad request response`() {
```

Comment:
I don't think we need this test case here - we should make clientId a mandatory field on the request and remove the validation annotations